src/dwfl_symbol_lookup.cc

```diff
@@ -128,6 +130,23 @@ SymbolIdx_t DwflSymbolLookup::insert(const DDProfMod &ddprof_mod,
     assert(0);
   }
 
+  // previous element was not within bounds, but it still matches
```

Comment:
As discussed, it would be better to understand why the size of the symbol is returned as 0